[18.0][MIG] stock_picking_report_delivery_cost: Migration to 18.0

BhaveshHeliconia opened the pull request.

pedrobaeza committed:
[ADD] stock_picking_report_delivery_cost: Delivery cost in Stock Picking Report

This module shows in the delivery slip and picking operations reports the cost
of the delivery method according the rate computed on the fly according same
conditions as standard `delivery` module would do.

Usage
=====

To use this module, you need to:

* Go to *Inventory > Operations > Transfers* and pick one not in state *Done*
  or *Cancelled*.
* Click on *Print > Delivery Slip* or *Print > Picking Operations*.
* You will see in the report the computed delivery cost.

pedrobaeza committed:
[MIG] stock_picking_report_delivery_cost: Migration to 12.0

* Standard procedure
* Tests adapted

[UPD] Update stock_picking_report_delivery_cost.pot

victoralmau committed:
[IMP] stock_picking_report_delivery_cost: black, isort, prettier

victoralmau committed:
[MIG] stock_picking_report_delivery_cost: Migration to 13.0

[UPD] Update stock_picking_report_delivery_cost.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-13.0/delivery-carrier-13.0-stock_picking_report_delivery_cost
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-13-0/delivery-carrier-13-0-stock_picking_report_delivery_cost/

pilarvargas-tecnativa committed:
[MIG] stock_picking_report_delivery_cost: Migration to version 15.0

TT39015

[UPD] Update stock_picking_report_delivery_cost.pot

[UPD] README.rst

[UPD] README.rst

victoralmau committed:
[MIG] stock_picking_report_delivery_cost: Migration to 16.0

TT44367

Commit:
[UPD] Update stock_picking_report_delivery_cost.pot

kbentaleb committed:
Added translation using Weblate (French)

kbentaleb committed:
Translated using Weblate (French)

Currently translated at 75.0% (3 of 4 strings)

Translation: delivery-carrier-16.0/delivery-carrier-16.0-stock_picking_report_delivery_cost
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-16-0/delivery-carrier-16-0-stock_picking_report_delivery_cost/fr/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: delivery-carrier-16.0/delivery-carrier-16.0-stock_picking_report_delivery_cost
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-16-0/delivery-carrier-16-0-stock_picking_report_delivery_cost/it/

RodrigoBM committed:
[FIX] stock_picking_report_delivery_cost: add the same class style as in the base report

Commit:
[UPD] Update stock_picking_report_delivery_cost.pot

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: delivery-carrier-16.0/delivery-carrier-16.0-stock_picking_report_delivery_cost
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-16-0/delivery-carrier-16-0-stock_picking_report_delivery_cost/it/

BhaveshHeliconia committed:
[IMP] stock_picking_report_delivery_cost: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] stock_picking_report_delivery_cost: Migration to 18.0

pedrobaeza commented:
/ocabot migration stock_picking_report_delivery_cost

victoralmau commented on stock_picking_report_delivery_cost/models/stock_picking.py:
```diff
@@ -0,0 +1,27 @@
+    _inherit = "stock.picking"
+
+    # Needed for fields.Monetary
+    currency_id = fields.Many2one(
+        related="sale_id.currency_id",
+        readonly=True,
+        string="Currency",
+        related_sudo=True,  # for avoiding access problems
+    )
```
This is incorrect (an error occurs) if we try to print a picking that is not linked to a sales order (`sale_id`).

```suggestion
    currency_id = fields.Many2one(
        comodel_name="res.currency",
        string="Currency",
        compute="_compute_currency_id",
        compute_sudo=True,  # for avoiding access problems
    )
    
    @api.depends("sale_id", "sale_id.currency_id", "company_id")
    def _compute_currency_id(self):
        for item in self:
            item.currency_id = item.sale_id.currency_id or item.company_id.currency_id
```

victoralmau committed:
[FIX] stock_picking_report_delivery_cost: Fix currency_id field + Improve tests

Related to https://github.com/OCA/delivery-carrier/pull/965#discussion_r2242004270

If a picking is created manually (not linked to a sales order) and has a defined price (carrier_price), the report should not generate an error.

BhaveshHeliconia commented:
@victoralmau You’re right—I tested it and updated the implementation as you suggested. Thanks for the helpful input!

victoralmau reviewed (approved):
Functional review OK

TT54520

victoralmau commented:
Although everything is correct, add the commit https://github.com/OCA/delivery-carrier/pull/1040/commits/4f9a6372fa8406bf0d9cf29868915867c68c29b5 from 16.0 before the pre-commit commit so that the change history is consistent between versions.

victoralmau reviewed (changes requested):
The `_compute_currency_id()` method appears twice because the changes from https://github.com/OCA/delivery-carrier/commit/4f9a6372fa8406bf0d9cf29868915867c68c29b5 are added after the migration commit, and the migration commit also partially adds those changes.

The solution is to add the https://github.com/OCA/delivery-carrier/commit/4f9a6372fa8406bf0d9cf29868915867c68c29b5 commit **before** the pre-commit commit.
<img width="898" height="510" alt="ejemplo-github" src="https://github.com/user-attachments/assets/e7a39e8c-77a5-48d1-b32e-3c7f98142ad7" />

BhaveshHeliconia commented:
@victoralmau : Please review and let me know if I’m still missing anything.

victoralmau commented on stock_picking_report_delivery_cost/tests/test_stock_picking_report_delivery_cost.py:
```diff
@@ -52,9 +49,6 @@ def setUpClass(cls):
         )
-        cls.warehouse = cls.env["stock.warehouse"].search(
-            [("company_id", "=", cls.env.company.id)], limit=1
-        )
```
You should not delete this.

victoralmau commented on stock_picking_report_delivery_cost/tests/test_stock_picking_report_delivery_cost.py:
```diff
@@ -79,25 +73,6 @@ def test_carrier_price_for_report_after(self):
-        res = self.env["ir.actions.report"]._render(
-            "stock.report_deliveryslip", picking.ids
-        )
-        self.assertRegex(str(res[0]), picking.name)
```
You should not delete this.

victoralmau commented on stock_picking_report_delivery_cost/models/stock_picking.py:
```diff
@@ -8,10 +8,10 @@ class StockPicking(models.Model):
         string="Currency",
-        related_sudo=True,  # for avoiding access problems
+        compute="_compute_currency_id",
+        compute_sudo=True,  # for avoiding access problems
```
These changes are added in the migration commit, but they should be in the “**[FIX] stock_picking_report_delivery_cost: Fix currency_id field + Improve tests**” commit right?

BhaveshHeliconia commented:
@victoralmau : Now i think all fixed.

pedrobaeza commented on stock_picking_report_delivery_cost/report/report_shipping.xml:
```diff
@@ -6,10 +6,10 @@
                 name="div_carrier_price_for_report"
             >
-                <strong>Delivery cost:</strong>
+                <strong>Delivery cost</strong>
```
This change is making all the current translations invalid. Is it needed? If so, please adapt all the translated PO files to remove the `:` both in the source and the translation.

BhaveshHeliconia commented:
@pedrobaeza : I’ve updated all the translation files as suggested. Please review and let me know if the changes are correct or if anything needs adjustment.

pedrobaeza reviewed (approved):
/ocabot merge nobump
